app/views/hub/documents/index.html.erb

```diff
@@ -37,6 +40,10 @@
           <td class="index-table__cell">
             <%= document.tax_return&.year %>
           </td>
+          <td class="index-table__cell">
+            <% uploaded_by_name = document.uploaded_by_type == 'User' ? document.uploaded_by&.name : document.uploaded_by&.legal_name || document.uploaded_by&.preferred_name %>
```

Comment:
This would be really nice to put in a helper so that we can keep logic in the view pretty simple. Also, instead of putting the legal name or preferred name for the client, we might be able to get away with just putting "Client".

Reply:
Instead of looking directly at uploaded_by_type, you could type check: `document.uploaded_by.is_a? User`. I think that would be slightly more semantically "rubyish" :D

Reply:
Instead of looking directly at uploaded_by_type, you could type check: `document.uploaded_by.is_a? User`. I think that would be slightly more semantically "rubyish" :D